internal/controller/pathwaysjob_controller.go

```diff
@@ -0,0 +1,663 @@
+/*
+Copyright 2025.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"context"
+	"fmt"
+	"slices"
+	"strconv"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog/v2"
+	"k8s.io/utils/ptr"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	jobsetv1alpha2 "sigs.k8s.io/jobset/api/jobset/v1alpha2"
+	jobsetclient "sigs.k8s.io/jobset/client-go/clientset/versioned"
+
+	pathwaysjob "pathways-job/api/v1"
+)
+
+// PathwaysJobReconciler reconciles a PathwaysJob object
+type PathwaysJobReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+}
+
+// Public variables to store TPU information
+var (
+	TPUVersion   string
+	TPUTopology  string
+	InstanceType string
+	NumVMs       int32
+)
+
+// Map to convert worker type to Instance type
+var WorkerTypeToTPUVersionMap = map[string]string{
+	"tpu-v6e-slice":        "tpuv6e",
+	"tpu-v5p-slice":        "tpuv5",
+	"tpu-v5-lite-podslice": "tpuv5",
```

Comment:
This doesn't seem right, should it be `tpuv5e` or `tpuv5lite`

Reply:
Updated now, thank you for the catch!